xonsh/parsers/base.py

```diff
@@ -2453,8 +2470,13 @@ def p_string_literal(self, p):
                 "__xonsh__.path_literal", [s], lineno=p1.lineno, col=p1.lexpos
             )
         elif "f" in prefix:
-            s = eval_fstr_fields(p1.value, prefix, filename=self.lexer.fname)
-            s = pyparse(s).body[0].value
+            try:
+                s = pyparse(p1.value)
+            except SyntaxError:
+                s = None
+            if s is None:
+                s = eval_fstr_fields(p1.value, prefix, filename=self.lexer.fname)
```

Comment:
This can raise an exception. @scopatz, @gforsyth, how can I emit syntax error from within the parser code?

Reply:
you can call `self._parse_error(msg, self.currloc(lineno=p1.lineno, column=p1.lexpos))` for some message `msg`

Reply:
Thanks for your help @scopatz, but no, that doesn't do the trick. It's strange, but it doesn't do anything in fact.

Reply:
hmmm that really should raise a SyntaxError.  Maybe it is being caught somewhere?

Reply:
Yes, most probably yes. It seems to be broken for quite a long time. I don't remember ever getting a syntax error for `f"{}"`.